contracts/governance/KyberStaking.sol

```diff
@@ -0,0 +1,466 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+
+import {Math} from '@openzeppelin/contracts/math/Math.sol';
+import {SafeMath} from '@openzeppelin/contracts/math/SafeMath.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import {ReentrancyGuard} from '@openzeppelin/contracts/utils/ReentrancyGuard.sol';
+import {PermissionAdmin} from '@kyber.network/utils-sc/contracts/PermissionAdmin.sol';
+
+import {IKyberStaking} from '../interfaces/IKyberStaking.sol';
+import {EpochUtils} from '../misc/EpochUtils.sol';
+
+interface IWithdrawHandler {
+  function handleWithdrawal(address staker, uint256 reduceAmount) external;
+}
+
+/**
+ * @notice   This contract is using SafeMath for uint, which is inherited from EpochUtils
+ *           Some events are moved to interface, easier for public uses
+ */
+contract KyberStaking is IKyberStaking, EpochUtils, ReentrancyGuard, PermissionAdmin {
+  using Math for uint256;
```

Comment:
why do we need both Math and SafeMath here?

Reply:
Math for `min` function
SafeMath for `add` `sub` `mul` `divide`